.github/workflows/deploy-terraform.yaml

```diff
@@ -0,0 +1,35 @@
+---
+name: deploy-terraform
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+
+jobs:
+  deploy-terraform:
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint
+    # so that we can authenticate with AWS
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Terraform
+        uses: ./.github/actions/setup-terraform
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
+          terraform-iam-role-arn: ${{ secrets.TERRAFORM_IAM_ROLE_ARN }}
+          aws-account-id: ${{ secrets.AWS_ACCOUNT_ID }}
+
+      - name: Validate Terraform config
+        run: terraform validate
+        shell: bash
+
+      - name: Apply Terraform changes
+        run: terraform apply -auto-approve
```

Comment:
The [`apply` command](https://developer.hashicorp.com/terraform/cli/commands/apply) first diffs the local Terraform config defined in `main.tf` against the state of remote resources in AWS, and applies any necessary changes based on that diff (including creating, updating, or deleting resources)